fabric-fluid-v1/src/main/java/net/fabricmc/fabric/api/fluids/v1/container/volume/FluidVolume.java

```diff
@@ -0,0 +1,45 @@
+package net.fabricmc.fabric.api.fluids.v1.container.volume;
+
+import java.util.Iterator;
+
+import com.google.common.collect.Iterators;
+
+import net.minecraft.fluid.Fluid;
+import net.minecraft.nbt.CompoundTag;
+
+import net.fabricmc.fabric.api.fluids.v1.container.FluidContainer;
+
+/**
+ * a fluid container that can only hold 1 fluid.
+ */
+public interface FluidVolume extends FluidContainer {
+	@Override
+	default long getTotalVolume() {
+		return this.getAmount();
+	}
+
+	/**
+	 * @return the amount of fluid currently in the fluid volume.
+	 */
+	long getAmount();
+
+	@Override
+	default Iterator<FluidVolume> iterator() {
+		return Iterators.singletonIterator(this);
+	}
+
+	@Override
+	default FluidVolume simpleCopy() {
+		return new SimpleFluidVolume(this.getFluid(), this.getAmount(), this.getData() == null ? null : this.getData().copy());
+	}
+
+	/**
+	 * @return the id for the fluid
+	 */
+	Fluid getFluid();
+
+	/**
+	 * @return the data associated with the fluid
```

Comment:
please mention that this is nullable

Reply:
do docs even work in fapi yet?

Reply:
yes, @Nullable good

Reply:
yes it publishs the source jar and you have been using lower case everywhere and it hurts my eye

Reply:
did we ever decide what annotation api to use, or is that still in 🚲🏠

Reply:
@Nullable makes it become a nullable type in kotlin, use that

Reply:
just write it is nullable in the javadoc for now

Reply:
k

Reply:
also no annotations since we are still bikesheding on it

Reply:
how can you bikeshed on a compile time dependency

Reply:
yes

Reply:
fabric is magic, we are bikesheding on the license year what do you expect